### DISCUSSION TITLE
Funds transfer

### DISCUSSION THREAD

**Discussion by gumdropsteve** [gumdropsteve/dispute-resolution#10] [Ideas]
TLDR: having renter submit safety deposit for mvp implementation

Thought: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/escrow/Escrow.sol

## Issue
```
  *                                ... The contract that uses the escrow as its
  * payment method should be its owner...
```
so whoever opens is the person depositing

#### Questions
- placing pre-hold on funds seems lame

## Solutions
- opened by renter upon offer submission
  - when deposit? how much?
  - stake?
  - [RefundEscrow](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/escrow/RefundEscrow.sol)
    - 1 time thing? can payee be changed?
    - gas fees for what? because we assume i'm irresponsible?
- deposit covered by dao, defi, skrrr
  - trust first 

#### withdraw()
[withdraw()](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/escrow/Escrow.sol#L54) method [reentry](https://consensys.github.io/smart-contract-best-practices/known_attacks/#reentrancy) vulnerabilities.. [what?](https://quantstamp.com/blog/what-is-a-re-entrancy-attack)
```
     * WARNING: Forwarding all gas opens the door to reentrancy vulnerabilities.
     * Make sure you trust the recipient, or are either following the
     * checks-effects-interactions pattern or using {ReentrancyGuard}. 
```

#### depositor(s)
as is, `RefundEscrow` deposits can be made by multiple parties, single party payout... ([source](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/escrow/RefundEscrow.sol#L7-L16))
```
 * @dev Escrow that holds funds for a beneficiary, deposited from multiple
 * parties.
```
EDIT: Whoever deploys `Escrow` is the only 1 who can deposit, and the only 1 who can trigger withdraw? Multiple payees ok. Self as payee ok. [testing](https://rinkeby.etherscan.io/address/0x53f989d382D5c1A8B703e9774f71c3194f77F266#writeContract)

EDIT: Just make own, note [Address](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Address.sol) is used `for address payable;` in `Escrow`
